home/tests/test_api.py

```diff
@@ -244,118 +254,142 @@ def test_pagination(self):
             )
             body.append(("Whatsapp_Message", block_value))
 
-        self.content_page1.whatsapp_body = body
-        self.content_page1.save_revision().publish()
+        page1.whatsapp_body = body
+        page1.save_revision().publish()
 
         # it should only return the 11th paragraph if 11th message
         # is requested
-        response = self.client.get(
-            f"/api/v2/pages/{self.content_page1.id}/?whatsapp=True&message=11"
-        )
+        response = uclient.get(f"/api/v2/pages/{page1.id}/?whatsapp=True&message=11")
         content = json.loads(response.content)
-        self.assertEqual(content["body"]["message"], 11)
-        self.assertEqual(content["body"]["next_message"], 12)
-        self.assertEqual(content["body"]["previous_message"], 10)
-        self.assertEqual(content["body"]["text"]["value"]["message"], "WA Message 11")
+        assert content["body"]["message"] == 11
+        assert content["body"]["next_message"] == 12
+        assert content["body"]["previous_message"] == 10
+        assert content["body"]["text"]["value"]["message"] == "WA Message 11"
 
-    def test_number_of_queries(self):
-        with self.assertNumQueries(8):
-            self.client.get("/api/v2/pages/")
+    def test_number_of_queries(self, uclient, django_assert_num_queries):
+        """
+        FIXME:
+         * Is the number of queries significant? What does it depend on?
+         * Should this be part of some other test?
+        """
```

Comment:
I think the point of this is to ensure that we're not doing n+1 queries when listing the pages.

Usually for this kind of test, I run it to assert on an incorrect number of queries, which gives me a list of queries, that I can investigate. Then I leave a comment with a description of each of the queries, so that by looking at the test, someone can easily tell what each of the queries are.

Reply:
Ugh. This does two extra queries when it's the first test that runs.

Reply:
I have no idea what all these queries are doing, so I haven't tried to document them. 🤷

Reply:
Ah, I imagine that's internal wagtail things. This used to take a lot more queries before we enabled cache for testing, so if you want fix the "different number of queries if run first" issue you could disable cache for just this test and update the number of queries.

Reply:
Making the request twice and only counting the second sorts out the variance.  We probably want to measure this with caching enabled anyway, so I'll leave it as-is.